src/librustc_span/symbol.rs

```diff
@@ -1047,6 +1048,7 @@ pub mod kw {
 }
 
 // This module has a very short name because it's used a lot.
+#[allow(rustc::default_hash_types)]
```

Comment:
This annotation should work, all the allow/warn/deny hierarchy is built and checked after all macros are expanded.